plugins/modules/ipaconfig.py

```diff
@@ -0,0 +1,441 @@
+                                     aliases=['usersearch']),
+            ipagroupsearchfields=dict(type="list", required=False,
+                                      aliases=['groupsearch']),
+            ipamigrationenabled=dict(type="bool", required=False,
+                                     aliases=['enable-migration']),
+            ipagroupobjectclasses=dict(type="list", required=False,
+                                       aliases=['groupobjectclasses']),
+            ipauserobjectclasses=dict(type="list", required=False,
+                                      aliases=['userobjectclasses']),
+            ipapwdexpadvnotify=dict(type="int", required=False,
+                                    aliases=['pwdexpnotify']),
+            ipaconfigstring=dict(type="list", required=False,
+                                 choices=["AllowNThash",
+                                          "KDC:Disable Last Success",
+                                          "KDC:Disable Lockout",
+                                      "KDC:Disable Default Preauth for SPNs"]), # noqa E128
+            ipaselinuxusermaporder=dict(type="list", required=False),
+            ipaselinuxusermapdefault=dict(type="str", required=False),
+            ipakrbauthzdata=dict(type="list", required=False,
+                                 aliases=["pac-type"],
+                                 choices=["MS-PAC", "PAD", "nfs:NONE"]),
+            ipauserauthtype=dict(type="list", required=False,
+                                 aliases=["user-auth_type"]),
+            ipadomainresolutionorder=dict(type="list", required=False,
+                                          aliases=["domain-resolution_order"])
+        ),
+        supports_check_mode=True,
+    )
+
+    ansible_module._ansible_debug = True
+
+    # Get parameters
+
+    # general
+    ipaadmin_principal = module_params_get(ansible_module,
+                                           "ipaadmin_principal")
+    ipaadmin_password = module_params_get(ansible_module,
+                                          "ipaadmin_password")
+
+    params = {}
+    for x in [
+            "ipamaxusernamelength",
+            "ipahomesrootdir",
+            "ipadefaultloginshell",
+            "ipadefaultprimarygroup",
+            "ipadefaultemaildomain",
+            "ipasearchtimelimit",
+            "ipasearchrecordslimit",
+            "ipausersearchfields",
+            "ipagroupsearchfields",
+            "ipamigrationenabled",
+            "ipagroupobjectclasses",
+            "ipauserobjectclasses",
+            "ipapwdexpadvnotify",
+            "ipaconfigstring",
+            "ipaselinuxusermaporder",
+            "ipaselinuxusermapdefault",
+            "ipakrbauthzdata",
+            "ipauserauthtype",
+            "ipadomainresolutionorder"]:
+
+        val = module_params_get(ansible_module, x)
+        if val is not None:
+            params[x] = val
```

Comment:
It is preferred to use the CLI variable names than the internal names for the variables.
(If you took this from sudorule, is is probably my fault, sorry.)